Add MacPorts support with preconditions

ain opened the pull request:
Resolve #32 Alternatives to HomeBrew by MacPorts with preconditions:
- Boot2Docker
- Docker Compose

ain committed:
Add MacPorts support with preconditions

Resolve #32 Alternatives to HomeBrew by MacPorts with preconditions:
- Boot2Docker
- Docker Compose

brikis98 commented:
Thank you for the pull request. I like the general approach and I think this could be made to work, but as I mentioned in #32, before increasing the complexity of the installer, we need to find a way to add good automated tests, which is currently blocked by #7.

On a side note, this PR seems to have a lot of whitespace changes, which makes it hard to identify the real changes. Could you check your text editor settings for what could be causing that?

brikis98 commented on src/setup.sh:
```diff
@@ -273,7 +288,34 @@ function get_env_file {
+# Checks if all MacPorts requirements are met.
+#
+function check_port_prerequisites {
+  if ! type boot2docker > /dev/null 2>&1 ; then
```
Boot2Docker is one of the items in the `port_install` list. Why is it a prerequisite?

ain replied:
One of the reasons why a single install method is a bad idea and would grow
into conditional spaghetti: MacPorts has no Portfile for Boot2Docker or
Docker Compose.

brikis98 commented on src/setup.sh:
```diff
@@ -273,7 +288,34 @@ function get_env_file {
+#
+function check_port_prerequisites {
+  if ! type boot2docker > /dev/null 2>&1 ; then
+    log_error "MacPorts-based setup expects Docker Compose to be preinstalled."
```
Looks like a copy/paste error. This error message should be about Boot2Docker and the one below should be about Docker Compose.

brikis98 commented on src/setup.sh:
```diff
@@ -273,7 +288,34 @@ function get_env_file {
+  if ! type boot2docker > /dev/null 2>&1 ; then
+    log_error "MacPorts-based setup expects Docker Compose to be preinstalled."
+    exit 1
+  elif ! type docker-compose > /dev/null 2>&1 ; then
```
Is there no MacPorts Docker Compose recipe?

brikis98 commented on src/setup.sh:
```diff
@@ -294,19 +336,32 @@ function check_prerequisites {
+    log_info "Updating MacPorts"
+    sudo port selfupdate
+
+    port_install "virtualbox" "VirtualBox" "VBoxManage"
```
I'm a little concerned that most of the HomeBrew list is duplicated here, but not all. It makes it trickier to maintain, as you have to update two places (albeit, at least here they are close together), and it's not obvious why some of the items are in one list but not another. Although bash is terrible at data structures, I'm wondering if there is a way to create one to make the solution look more obvious. For example, in JavaScript, I might have used an object like this:

``` js
var dependencies = [
  {packageName: "virtualBox", readableName: "Virtual Box", commandName: "VBoxManage", brew: false, brewCask: true, ports: true},
  {packageName: "docker-compose", readableName: "Docker Compose", commandName: "docker-compose", brew: true, brewCask: false, ports: false},
  // ...
]
```

ain replied:
Good call, but you have to consider possibly different naming conventions
as well.

brikis98 replied:
True. Perhaps this format is more like it:

``` js
var dependencies = [
  {readableName: "Virtual Box", commandName: "VBoxManage", brewName: null, brewCaskName: "virtualbox", portsName: "virtualbox"},
  {readableName: "Docker Compose", commandName: "docker-compose", brewName: "docker-compose", brewCaskName: null, portsName: null},
  // ...
]
```

ain replied:
Yup. If nesting is doable, I'd go Managers > Brew|Ports > PackageName,
inside a package descriptor, so you can add others later, e.g. pip or
whatever.

ain committed:
Fix erroneous MacPorts package check

Package was incorrectly reported to be installed.
grepped port list command avoids such behaviour.

ain committed:
Add improved port detection regex

ain committed:
Merge recent changes from brikis98/docker-osx-dev

ain committed:
Update README from brikis98/docker-osx-dev

ain commented:
Dumping maintenance on this PR. Diverged.
